app/src/main/kotlin/com/saltedge/authenticator/features/main/MainActivity.kt

```diff
@@ -21,263 +21,172 @@
-                authorizationId = authorizationID
-            )
-        )
-    }
-
-    override fun onNewAuthorization(authorizationIdentifier: AuthorizationIdentifier) {
-        this.addFragment(
-            AuthorizationDetailsFragment.newInstance(
-                connectionId = authorizationIdentifier.connectionID,
-                authorizationId = authorizationIdentifier.authorizationID
-            )
-        )
-    }
-
-    override fun closeView() {
-        finish()
-    }
-
-    override fun popBackStack() {
-        supportFragmentManager.popBackStack()
-    }
-
-    override fun showNoConnectionsError() {
-        val snackbar = this.buildWarning(
-            text = getString(R.string.connections_list_no_connections),
-            snackBarDuration = 5000
-        )
-        snackbar?.show()
-    }
-
-    override fun showSubmitActionFragment(
-        connectionGuid: GUID,
-        actionAppLinkData: ActionAppLinkData
-    ) {
-        this.addFragment(
-            SubmitActionFragment.newInstance(
-                connectionGuid = connectionGuid,
-                actionAppLinkData = actionAppLinkData
-            )
-        )
-    }
-
-    override fun showConnectionsSelectorFragment(connections: List<ConnectionViewModel>) {
-        this.addFragment(SelectConnectionsFragment.newInstance(connections = connections))
-    }
-
-    override fun updateNavigationViewsContent() {
-        isTopNavigationLevel().also { isOnTop ->
-            toolbarView?.navigationIcon =
-                ((currentFragmentInContainer() as? UpActionImageListener)?.getUpActionImageResId()
-                    ?: presenter.getNavigationIcon(isOnTop))?.let { resId ->
-                    this.getDrawable(resId)
-                }
-
-            bottomNavigationLayout?.setVisible(show = isOnTop)
-        }
+        viewModel.onViewClick(v?.id ?: return)
     }
 
     override fun getUnlockAppInputView(): UnlockAppInputView? = unlockAppInputView
 
     override fun getAppBarLayout(): View? = appBarLayout
 
-    override fun onBackStackChanged() {
-        presenter.onFragmentBackStackChanged(isTopNavigationLevel(), intent)
-    }
-
-    override fun getSnackbarAnchorView(): View? = snackBarCoordinator
-
-    override fun onNetworkConnectionChanged(isConnected: Boolean) {
-        showNetworkMessage(isConnected)
-    }
-
-    override fun onConnectionSelected(connectionGuid: String) {
-        presenter.onConnectionSelected(connectionGuid)
-    }
-
-    private fun showNetworkMessage(isConnected: Boolean) {
-        if (isConnected) {
-            snackbar?.dismiss()
-        } else {
-            snackbar = this.buildWarning(getString(R.string.warning_no_internet_connection))
-            snackbar?.show()
-        }
+    override fun getSnackbarAnchorView(): View? = container
+
+    private fun setupViewModel() {
+        viewModel = ViewModelProvider(this, viewModelFactory).get(MainActivityViewModel::class.java)
```

Comment:
Also u can use this solution(it is not deprecated ;) )
```  
      viewModel = ViewModelProviders
            .of(this, viewModelFactory)
            .get(MainActivityViewModel::class.java)
```

Reply:
ViewModelProviders is deprecated in the last lib version